### PR TITLE
DOC: sparse.linalg.gcrotmk : Typeset code objects in docstring

### PR DESCRIPTION
**Pull request by anushkasuyal**
Corrected the typesetting of code objects in the gcrotmk documentation. Ensured that 'A' in the parameter description for 'M' and 'callback(xk)' in the description of 'callback' are properly formatted as code.

### CODE REVIEW

File: scipy/sparse/linalg/_isolve/_gcrotmk.py [hunk 1]
```diff
@@ -205,14 +205,14 @@ def gcrotmk(A, b, x0=None, *, rtol=1e-5, atol=0., maxiter=1000, M=None, callback
...
-        inverse of A. gcrotmk is a 'flexible' algorithm and the preconditioner
+        Preconditioner for ``A``.  The preconditioner should approximate the
+        inverse of ``A``. gcrotmk is a 'flexible' algorithm and the preconditioner
         can vary from iteration to iteration. Effective preconditioning
         dramatically improves the rate of convergence, which implies that
         fewer iterations are needed to reach a given error tolerance.
     callback : function, optional
         User-supplied function to call after each iteration.  It is called
-        as callback(xk), where xk is the current solution vector.
+        as ``callback(xk)``, where xk is the current solution vector.
```
Comment (lucascolley) [inline scipy/sparse/linalg/_isolve/_gcrotmk.py:215]:
```suggestion
        as ``callback(xk)``, where ``xk`` is the current solution vector.
```

**Review by lucascolley — COMMENTED** [review]
thanks @anushkasuyal ! The double backticks are consistent with the rest of the docstring, so this is an improvement 👍 added one more suggestion

**Comment by j-bowhay** [comment]
> thanks @anushkasuyal ! The double backticks are consistent with the rest of the docstring, so this is an improvement 👍 added one more suggestion

Whilst I agree they are consistent, I think A should be surrounded by single backticks as it is an argument to the function.

**Comment by lucascolley** [comment]
>  I think A should be surrounded by single backticks as it is an argument to the function.

In which case there are a few other areas of the docstring to change

**Review by j-bowhay — CHANGES_REQUESTED** [review]
Thanks @anushkasuyal can you change ` ``A`` ` to `` `A` `` throughout this docstring as per https://numpydoc.readthedocs.io/en/latest/format.html#parameters

**Comment by lucascolley** [comment]
superseded by gh-21094

**Pull request closed without merging**